smart_contracts/vm2/macros/src/lib.rs

```diff
@@ -1311,6 +1323,31 @@ fn casper_trait_definition(mut item_trait: ItemTrait, trait_meta: TraitMeta) ->
 
                                 #handle_ret
                             }
+
+                            #[cfg(target_arch = "wasm32")]
+                            #[no_mangle]
+                            #vis extern "C" fn #dispatch_func_name<T>()
```

Comment:
Now that I look at it, since it's a generic function, we _may_ not need `#[no_mangle] extern "C"` here. When we bind this generic dispatch method to a concrete type (in `impl Foo for Bar`), then it has to be `extern "C"`. Not critical, just may simplify by having one copy of `#dispatch_func_name` without extern "C".